packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2660,6 +2652,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                         error: `lastSequenceNumber changed before uploading to storage. ${this.deltaManager.lastSequenceNumber} !== ${summaryRefSeqNum}`,
                     };
                 }
+                assert(summaryRefSeqNum === this.deltaManager.lastMessage?.sequenceNumber,
+                    "it's one and the same thing");
+
+                if (lastAck !== this.summaryCollection.latestAck) {
```

Comment:
>lastAck !== this.summaryCollection.latestAck)  [](http://example.com/codeflow?start=20&length=46)

I tried to see a scenario where it could happen and could not :) Do you see a scenario where it can possibly happen ?